### ISSUE TITLE
Rule Change: Make no-html rule's allowed option case-insensitive

### ISSUE THREAD

**Issue by TKDev7** [eslint/markdown#479]
### What rule do you want to change?

no-html

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
<div></div>
<DIV></DIV>
```

### What does the rule currently do for this code?

Currently, if `allowed: ["DIV"]` is configured, the rule will correctly allow `<DIV></DIV>`. However, `<div></div>` will be reported as an error because the matching is case-sensitive. This leads to false positives since HTML tags are, by specification, case-insensitive.

### What will the rule do after it's changed?

After this change, the no-html rule will perform case-insensitive matching when checking HTML tag names against the `allowed` option list. This means that if `allowed: ["DIV"]` is configured, both `<div></div>` and `<DIV></DIV>` (and any other casing like `<DiV>`) will be correctly recognized as allowed and will not be reported as errors.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by snitin315** [comment]
Since HTML tags are by default case insensitive, I think we can just provide `allowIgnoreCase: boolean` 👍🏻. Will mark this accepted if the team agrees on the same.

cc @JoshuaKGoldberg @lumirlumir

**Comment by snitin315** [comment]
@TKDev7 We generally use present tense. However in this case the option was already named as `allowed` so we don't change it once it's done. 

I don't have a strong opinion on the name, I'm fine with whatever the team decides `allowIgnoreCase`  or `allowedIgnoreCase`

**Comment by lumirlumir** [comment]
I know ESLint usually uses `allow` and `allowXXX` for array or boolean options, but since the preceding option name is `allowed`, I think `allowedIgnoreCase` works well here.

So, let's go with `allowedIgnoreCase: boolean` for this case 👍

**Comment by SwetaTanwar** [comment]
@TKDev7 Can I take this up since you are already occupied with other issues & PRs?
